code/datums/elements/cliff_walker.dm

```diff
@@ -0,0 +1,33 @@
+/// Lets a mob walk cliffs and keeps track of if they're alive or not to add/remove the trait
+/datum/element/cliff_walking
+
+/datum/element/cliff_walking/Attach(datum/target, climb_time, climb_stun)
+	. = ..()
+
+	if(!isliving(target))
+		return ELEMENT_INCOMPATIBLE
+
+	// Feel free to add more bespoke signals here if this gets implemented for more than just a few funny mobs
+	RegisterSignals(target, list(COMSIG_LIVING_DEATH, COMSIG_LIVING_REVIVE), PROC_REF(update_cliff_walking))
+
+	update_cliff_walking(target)
+
+/datum/element/cliff_walking/Detach(datum/source, ...)
+	. = ..()
+
+	UnregisterSignal(source, list(COMSIG_LIVING_DEATH, COMSIG_LIVING_REVIVE))
+
+/// Do some checks to see if we should walk the cliffs
+/datum/element/cliff_walking/proc/update_cliff_walking(mob/living/climber)
+	SIGNAL_HANDLER
+
+	if(climber.stat == DEAD)
+		REMOVE_TRAIT(climber, TRAIT_CLIFF_WALKER, src)
```

Comment:
```suggestion
		REMOVE_TRAIT(climber, TRAIT_CLIFF_WALKER, type)
```
Sources are strings, I don't know what just passing `src` does here but it seems like a risky idea.
`type` or `REF(src)` are both approaches we have used but for an element it makes sense to me to use `type` because there won't be several instances of the element and it's more human-readable if you need to debug where a trait is coming from.

Alternately just make a trait source define for this, for the same reason of human readibility when debugging.

Reply:
I think just 'type' is probably most straightforward